src/config.rs

```diff
@@ -0,0 +1,55 @@
+//! This config contains all structs, enums and functions to configure MLS.
+//!
+
+use crate::codec::{Codec, CodecError};
+use crate::errors::ConfigError;
+
+/// # Protocol Version
+///
+/// 7. Key Packages
+///
+/// ```text
+/// enum {
+///     reserved(0),
+///     mls10(1),
+///     (255)
+/// } ProtocolVersion;
+/// ```
+///
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u8)]
+pub enum ProtocolVersion {
+    Reserved = 0,
+    Mls10 = 1,
+}
+
+/// There's only one version right now, which is the default.
+impl Default for ProtocolVersion {
+    fn default() -> Self {
+        ProtocolVersion::Mls10
+    }
+}
+
+impl Codec for ProtocolVersion {
+    fn encode(&self, buffer: &mut Vec<u8>) -> Result<(), CodecError> {
+        (*self as u8).encode(buffer)?;
+        Ok(())
+    }
+}
+
+impl ProtocolVersion {
+    /// Convert an integer to the corresponding protocol version.
+    ///
+    /// Returns an error if the protocol version is not supported.
+    pub fn from(v: u8) -> Result<ProtocolVersion, ConfigError> {
+        match v {
+            1 => Ok(ProtocolVersion::Mls10),
+            _ => Err(ConfigError::UnsupportedMlsVersion),
+        }
+    }
+
+    /// Returns a list of all supported protocol versions.
+    pub fn supported() -> Vec<Self> {
+        vec![Self::Mls10]
+    }
+}
```

Comment:
Not a blocker, but having an `is_supported()` function might be neat.

Reply:
yeah, that's a separate issue and should be tackled as part of #85